src/sfsecrethelper.cpp

```diff
@@ -0,0 +1,135 @@
+
+SFSecretHelper::SFSecretHelper(QString collection, QObject *parent)
+        : QObject{parent}
+        , m_collectionName{collection}
+{}
+
+void SFSecretHelper::instantiate(const QString &collection, QObject *parent) {
+    if (instance == nullptr) {
+        instance = new SFSecretHelper(collection, parent);
+    }
+}
+
+SFSecretHelper & SFSecretHelper::getInstance() {
+    return *instance;
+}
+
+QObject * SFSecretHelper::provider(QQmlEngine *engine, QJSEngine *scriptEngine) {
+    Q_UNUSED(engine)
+    Q_UNUSED(scriptEngine)
+
+    return instance;
+}
+
+
+void SFSecretHelper::store(const QString &key, const QByteArray &value)
+{
+    const Secret::Identifier identifier = this->generateIdentifier(key);
+
+    Secret secret(identifier);
+    secret.setData(value);
+
+    CreateCollectionRequest request;
+    request.setManager(&m_secretManager);
+    request.setCollectionLockType(CreateCollectionRequest::DeviceLock);
+    request.setDeviceLockUnlockSemantic(SecretManager::DeviceLockKeepUnlocked);
+    request.setAccessControlMode(SecretManager::OwnerOnlyMode);
+    request.setUserInteractionMode(SecretManager::SystemInteraction);
+    request.setCollectionName(identifier.collectionName());
+    request.setStoragePluginName(identifier.storagePluginName());
+    request.setEncryptionPluginName(identifier.storagePluginName());
+    request.startRequest();
+    request.waitForFinished();
+
+    if (request.result().code() != Sailfish::Secrets::Result::Succeeded) {
+        qWarning() << "Failed to create collection " << m_collectionName << " for storing the key!";
+        qWarning() << "Error:" << request.result().errorCode()
+                   << request.result().errorMessage();
+        emit finished(1, value);
+        return;
+    }
+
+    StoreSecretRequest storeRequest;
+    storeRequest.setManager(&m_secretManager);
+    storeRequest.setSecretStorageType(StoreSecretRequest::CollectionSecret);
+    storeRequest.setUserInteractionMode(SecretManager::SystemInteraction);
+    storeRequest.setSecret(secret);
+    storeRequest.startRequest();
+    storeRequest.waitForFinished();
+
+    if (storeRequest.result().code() != Sailfish::Secrets::Result::Succeeded) {
+        qWarning() << "Failed to store key: "<< key << " !";
+        qWarning() << "Error:" << storeRequest.result().errorCode()
+                   << storeRequest.result().errorMessage();
+        emit finished(1, value);
+        return;
+    }
+    //Success
+    emit finished(0);
+}
+
+void SFSecretHelper::get(const QString &key)
+{
+    const Secret::Identifier identifier = this->generateIdentifier(key);
+
+    StoredSecretRequest request;
+    request.setManager(&m_secretManager);
+    request.setUserInteractionMode(Sailfish::Secrets::SecretManager::SystemInteraction);
+    request.setIdentifier(identifier);
+    request.startRequest();
+    request.waitForFinished();
+
+    if (request.result().code() != Sailfish::Secrets::Result::Succeeded)
+    {
```

Comment:
I'll do. You don't use clang-format or any other codestyle checker, do you?

Reply:
I'm sorry, I don't have anything like this set up. I'll see if I can put something together because it would certainly be a good idea (especially given you identified places where I've failed to follow my own requirements elsewhere!).

At the very least, I should have had a style guide written down somewhere, and I completely understand there was no way for you to know about these things without it. The general rule, of course, it's to follow the surrounding code where you can.